scanpipe/pipes/purldb.py

```diff
@@ -320,3 +325,74 @@ def populate_purldb_with_discovered_dependencies(project, logger=logger.info):
         chunk_size=10,
         logger=logger,
     )
+
+
+def send_project_json_to_matchcode(
+    project_json_location, timeout=DEFAULT_TIMEOUT, api_url=PURLDB_API_URL
+):
+    """
+    Given a path to a ScanCode.io Project JSON output, `project_json_location`,
+    send the contents of the JSON file to PurlDB for matching.
+    """
+    project_json_contents = open(project_json_location, "rb")
+    files = {"upload_file": project_json_contents}
+
+    response = request_post(
+        url=f"{api_url}matching/",
+        data={},
+        timeout=timeout,
+        files=files,
+    )
+
+    return response
+
+
+def match_to_purldb(project):
+    """
+    Given a `project` where `scan_output_location` has been set, send the scan
+    of the project to PurlDB for matching. When available, process match results
+    by DiscoveredPackges from the matched package data.
+    """
+    from scanpipe.pipes.d2d import create_package_from_purldb_data
+
+    # send scan to purldb
+    response = send_project_json_to_matchcode(project.scan_output_location)
+    run_url = response["runs"][0]["url"]
+    url = response.get("url")
+    results_url = url + "results/"
+
+    # poll and see if the match run is ready
+    while True:
+        response = request_get(run_url)
+        if response:
+            status = response["status"]
+            if status == "success":
+                break
+        time.sleep(10)
```

Comment:
Make this a reusable function, something like: `def poll_until_success(url, sleep=10)`